SPM Example/PortalSwift/ViewController.swift

```diff
@@ -220,29 +230,151 @@ class ViewController: UIViewController, UITextFieldDelegate {
       throw PortalExampleAppError.userNotLoggedIn()
     }
     guard let config else {
-      self.logger.error("ViewController.recover() - ❌ Application configuration not set.")
+      self.logger.error("ViewController.eject() - ❌ Application configuration not set.")
       throw PortalExampleAppError.configurationNotSet()
     }
-    guard let cipherTextUrl = URL(
-      string: "\(config.custodianServerUrl)/mobile/\(user.exchangeUserId)/cipher-text/fetch?backupMethod=\(withBackupMethod.rawValue)"
-    ) else {
-      throw URLError(.badURL)
+
+    var cipherText: String? = nil
+    var organizationShare: String? = nil
+
+    guard let client = try await portal.client else {
+      throw PortalExampleAppError.clientInformationUnavailable("No client found.")
     }
-    let cipherTextData = try await requests.get(cipherTextUrl)
-    let cipherTextResponse = try decoder.decode(CipherTextResult.self, from: cipherTextData)
 
-    guard let organizationBackupShareUrl = URL(
-      string: "\(config.custodianServerUrl)/mobile/\(user.exchangeUserId)/org-share/fetch?backupMethod=\(withBackupMethod.rawValue)-SECP256K1"
-    ) else {
-      throw URLError(.badURL)
+    let backupWithPortal = client.environment?.backupWithPortalEnabled ?? false
+
+    if !backupWithPortal {
+      guard let cipherTextUrl = URL(
+        string: "\(config.custodianServerUrl)/mobile/\(user.exchangeUserId)/cipher-text/fetch?backupMethod=\(withBackupMethod.rawValue)"
+      ) else {
+        throw URLError(.badURL)
+      }
+      let cipherTextData = try await requests.get(cipherTextUrl)
+      let cipherTextResponse = try decoder.decode(CipherTextResult.self, from: cipherTextData)
+
+      cipherText = cipherTextResponse.cipherText
+
+      guard let organizationBackupShareUrl = URL(
+        string: "\(config.custodianServerUrl)/mobile/\(user.exchangeUserId)/org-share/fetch?backupMethod=\(withBackupMethod.rawValue)-SECP256K1"
+      ) else {
+        throw URLError(.badURL)
+      }
+      let organizationBackupShareData = try await requests.get(organizationBackupShareUrl)
+      let organizationBackupShareResponse = try decoder.decode(OrgShareResult.self, from: organizationBackupShareData)
+
+      organizationShare = organizationBackupShareResponse.orgShare
```

Comment:
Getting the `ED25519` organization share is missing.

We can change this code to be:

`
let secp256k1OrgShare = try await getOrgShare(custodianServerUrl: config.custodianServerUrl, exchangeUserId: user.exchangeUserId, method: withBackupMethod, curve: "SECP256K1").orgShare
        let ed2551OrgShare = try await getOrgShare(custodianServerUrl: config.custodianServerUrl, exchangeUserId: user.exchangeUserId, method: withBackupMethod, curve: "ED25519").orgShare
`
The above code might need to move the variables declaration to be out side of the if condition.

And add this private function:
`
    private func getOrgShare(custodianServerUrl: String, exchangeUserId: Int, method: BackupMethods, curve: String) async throws -> OrgShareResult {
        guard let organizationBackupShareUrl = URL(
          string: "\(custodianServerUrl)/mobile/\(exchangeUserId)/org-share/fetch?backupMethod=\(method.rawValue)-\(curve)"
        ) else {
          throw URLError(.badURL)
        }
        let organizationBackupShareData = try await requests.get(organizationBackupShareUrl)
        return try decoder.decode(OrgShareResult.self, from: organizationBackupShareData)
    }
`